packages/core/src/sharedWishlists/redux/middlewares/getWishlistSetsUponSharedWishlistIdChanges.js

```diff
@@ -0,0 +1,35 @@
+import {
+  CREATE_SHARED_WISHLIST_SUCCESS,
+  REMOVE_SHARED_WISHLIST_SUCCESS,
+} from '../actionTypes';
+import { doGetWishlistSets } from '../../../wishlists/redux/actions';
+import { getUser, getUserIsGuest } from '../../../entities/redux/selectors';
+import { getWishlistsSets as getWishlistsSetsClient } from '../../../wishlists/client';
+
+const getWishlistSets = doGetWishlistSets(getWishlistsSetsClient);
+
+/**
+ * Middleware to get the updated wishlist sets if a shared wishlist has been created or deleted.
+ *
+ * @function getWishlistSetsUponSharedWishlistIdChanges
+ * @memberof module:sharedWishlists/middlewares
+ *
+ * @param {object} store - Redux store at the moment.
+ *
+ * @returns {Function} Redux middleware.
+ */
+export default store => next => action => {
+  if (
+    action.type === REMOVE_SHARED_WISHLIST_SUCCESS ||
```

Comment:
Just to understand: If a shared wishlist is removed, why do we need to refetch the wishlist sets?

Reply:
We need to refetch when a shared wishlist is removed in case we want to immediately share it again. This way we warrantee we have the latest value for the `sharedWishlistId` in the wishlist set.

Reply:
Ok, thanks!